modules/oxide-components/src/main/ts/components/dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,77 @@
+import { useCallback, useEffect, useId, useRef, useState, type FC, type PropsWithChildren } from 'react';
+
+import { Button } from '../button/Button';
+
+import { DropdownContext, useDropdown } from './internals/context';
+import { getPositionStyles } from './internals/positioningUtils';
+
+export interface DropdownProps extends PropsWithChildren {
+  side?: 'top' | 'bottom' | 'left' | 'right';
+  align?: 'start' | 'center' | 'end';
+}
+
+// margin/gap between the trigger button and anchored container
+const GAP = 8;
+
+const Content: FC<PropsWithChildren> = ({ children, ...props }) => {
+  const { popoverId, triggerRef, side, align } = useDropdown();
+  const contentRef = useRef<HTMLDivElement>(null);
+
+  const [ positioningStyles, setPositioningStyles ] = useState({});
+
+  // this can be later replaced with CSS anchor positioning
+  const updatePosition = useCallback((event: Event) => {
+    if ((event as ToggleEvent).newState === 'open' && triggerRef.current && contentRef.current) {
+      const documentRect = document.documentElement.getBoundingClientRect();
+      const anchorRect = triggerRef.current.getBoundingClientRect();
+      const contentRect = contentRef.current.getBoundingClientRect();
+
+      // using document rect as a boundry, but maybe it should be the Editor area?
+      setPositioningStyles(getPositionStyles(anchorRect, contentRect, side, align, GAP, documentRect));
+    }
+  }, [ contentRef, triggerRef, align, side ]);
+
+  useEffect(() => {
+    const element = contentRef.current;
+    if (element === null) {
+      return;
+    }
+    element.addEventListener('toggle', updatePosition);
+
+    return () => {
+      element.removeEventListener('toggle', updatePosition);
+    };
+  }, [ contentRef, updatePosition ]);
+
+  // @ts-expect-error - TODO: Remove this expect error once we've upgraded to React 19+
+  return <div popover='auto' id={popoverId} ref={contentRef} { ...props } style={{
+    position: 'fixed',
+    overflow: 'auto',
+    display: '',
+    height: 'fit-content',
+    width: 'fit-content',
+    ...positioningStyles
+  }}>
+    {children}
+  </div>;
+};
+
+const TriggerButton: FC<PropsWithChildren> = ({ children }) => {
+  const { popoverId, triggerRef } = useDropdown();
+
+  // @ts-expect-error - TODO: Remove this expect error once we've upgraded to React 19+
+  return <Button variant='secondary' popovertarget={popoverId} popovertargetaction={'toggle'} ref={triggerRef} >{children}</Button>;
```

Comment:
I don't think this should always be a button, instead to should be only a wrapper that attaches the correct props. You can take some inspiration from [here](https://github.com/tinymce/tinymce-premium/blob/epic/TINY-12378/plugins/tinyai/src/main/ts/ui-react/components/Tooltip.tsx#L27-L62). Or we can set the popover api and then spread the props to the Button component